From e78b30af176e7a2f0307a45b0144749234cd10dc Mon Sep 17 00:00:00 2001
From: Ken Zhu <quic_guigenz@quicinc.com>
Date: Mon, 12 Aug 2024 17:35:35 -0700
Subject: [PATCH] net: fix callback trace issue when skb_debug_ojbect is
 enabled

In the linux-6.6, there are some new APIs added, it free
the skb to the skb cache directly without deactivate the
debug object.

Change-Id: I03516bfd0d389bbccb8de49b200bfe0a8703f4d1
Signed-off-by: Ken Zhu <quic_guigenz@quicinc.com>
---
 net/core/skbuff.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/net/core/skbuff.c b/net/core/skbuff.c
index ff0df949eaba..a083444ac635 100644
--- a/net/core/skbuff.c
+++ b/net/core/skbuff.c
@@ -1392,6 +1392,7 @@ kfree_skb_list_reason(struct sk_buff *segs, enum skb_drop_reason reason)
 
 		if (__kfree_skb_reason(segs, reason)) {
 			skb_poison_list(segs);
+			skbuff_debugobj_deactivate(segs);
 			kfree_skb_add_bulk(segs, &sa, reason);
 		}
 
@@ -1628,6 +1629,7 @@ static void napi_skb_cache_put(struct sk_buff *skb)
 	u32 i;
 
 	kasan_poison_object_data(skbuff_cache, skb);
+	skbuff_debugobj_deactivate(skb);
 	nc->skb_cache[nc->skb_count++] = skb;
 
 	if (unlikely(nc->skb_count == NAPI_SKB_CACHE_SIZE)) {
-- 
2.34.1

